app/logic/events.py

```diff
@@ -27,8 +25,8 @@ def getEvents(program_id=None):
 
     if program_id:
         Program.get_by_id(program_id) # raises an exception if program doesn't exist
-        return (Event.select(Event).join(ProgramEvent)
-                    .where(ProgramEvent.program == program_id).distinct())
+        return (Event.select(Event)
+                     .where(Event.program_id == program_id).distinct())
```

Comment:
No `.distinct()` is needed, as you aren't doing any joins. There aren't duplicate events in the `event` table. Also, you don't need to explicitly select `Event`  anymore.